### PR TITLE
MAX32666 platform drivers

### PR DESCRIPTION
**Pull request by CiprianRegus**
Added platform drivers support for MAX32666.

### CODE REVIEW

File: drivers/platform/maxim/maxim_spi.c [hunk 1]
```diff
@@ -56,6 +57,121 @@
...
+
+	/** This function will be a no-op for other chips, because this step is
+	 * done in the SDK in those cases.
+	*/
+#if TARGET_NUM != 32650 && TARGET_NUM != 32665
+	return 0;
+#endif
+
+	switch(id) {
+	case 0:
```
Comment (amiclaus) [inline drivers/platform/maxim/maxim_spi.c:78]: maybe we can use an enum with appropriate namings for these ids.

File: drivers/platform/maxim/maxim_uart.c [hunk 2]
```diff
@@ -265,30 +265,30 @@ int32_t no_os_uart_init(struct no_os_uart_desc **desc,
 		parity = MXC_UART_PARITY_DISABLE;
 		break;
 	case NO_OS_UART_PAR_MARK:
-#if TARGET_NUM == 32660 || TARGET_NUM == 32650
+#if TARGET_NUM == 32660 || TARGET_NUM == 32650 || TARGET_NUM == 32665
```
Comment (amiclaus) [inline drivers/platform/maxim/maxim_uart.c:268]: i'm wondering if there is an alternative to these ifdefs (we keep expanding them, and it starts to look a bit ugly :) )

**Review by amiclaus — COMMENTED** [review]
some small comments on my side.

File: drivers/platform/maxim/maxim_spi.c (hunk 1 above)
Comment (CiprianRegus) [reply drivers/platform/maxim/maxim_spi.c:78]: In this case, the `id` specifies the index of the SPI port (the `device_id` of a SPI descriptor), so I don't think that there is a point in creating a define for it.

**Review by adrimbarean — APPROVED** [review]
Looks good to me.

**Pull request merged**